analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Binary.cs

```diff
@@ -291,6 +291,31 @@ public void Binary_BoolOperands_Xor()
             SETestContext.CreateCS(code).Validator.ValidateTagOrder("True ^ True", "False ^ True", "True ^ False", "False ^ False");
         }
 
+        [TestMethod]
+        public void Binary_NoConstraint_VisitsBothBranches()
+        {
+            var code = """
+                if (boolParameter)
+                {
+                    Tag("If");
+                }
+                else
+                {
+                    Tag("Else");
+                }
+                Tag("End");
+                """;
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
+                "If",
+                "Else",
+                "End"); // FIXME, additional End missing
+            var symbol = validator.Symbol("boolParameter");
+            validator.TagStates("If").Should().ContainSingle().Which[symbol].Should().HaveNoConstraints();          // FIXME, True and NotNull are missing
+            validator.TagStates("Else").Should().ContainSingle().Which[symbol].Should().HaveNoConstraints();        // FIXME, False and NotNull are missing
+            validator.TagStates("End").Should().SatisfyRespectively(x => x[symbol].Should().HaveNoConstraints());   // FIXME, second branch and Bool- and NotNull constraints are missing
```

Comment:
@pavel-mikula-sonarsource Can you please have a look at this case (you may pick the unit test in a separate PR)? What I found so far: 
In `ProcessBranchState` the parameter symbol gets removed because `lva.LiveOut(branch.Source).ToHashSet()` returns 0 symbols. I think the parameter symbol should be in `liveVariables` but I'm not sure.
```cs
            var liveVariables = lva.LiveOut(branch.Source).ToHashSet();
            return state.RemoveSymbols(x => lva.IsLocal(x) && (x is ILocalSymbol or IParameterSymbol { RefKind: RefKind.None }) && !liveVariables.Contains(x))
                .ResetOperations();
```

Reply:
This is expected, because `boolParameter` is not used anymore,so we throw away it's state. We will never read it.

You need to use it `Tag("End", boolParameter)` or you need to inject `new PreserveTestCheck("boolParameter")` to explicitly tell the engine not to throw it away.

Reply:
In general, this UT should be writen using `Tag("If", boolParameter)` approach, with 2x `validator.ValidateTag` and 1x `validator.TagValues("End").Should().SatisfyRespectively(...)` instead